parser-typechecker/unison/Main.hs

```diff
@@ -164,16 +201,32 @@ runTranscripts inFork mcodepath args = do
                                          (FP.takeExtension md)
               writeUtf8 out mdOut
               putStrLn $ "💾  Wrote " <> out
-        wat -> putStrLn $ "Unrecognized command, skipping: " <> wat
-      PT.putPrettyLn $
+        wat -> 
+              PT.putPrettyLn $ P.callout "❓" (
+                P.lines [
+                  P.indentN 2 "Unrecognized command, skipping:", "",
+                  P.indentN 2 $ P.string wat])
+      pure True
+    [] -> 
+      pure False
+
+runTranscripts :: Bool -> Bool -> Maybe FilePath -> [String] -> IO ()
+runTranscripts inFork keepTemp mcodepath args = do
+  transcriptDir <- prepareTranscriptDir inFork mcodepath
+  completed <- runTranscripts' mcodepath transcriptDir args
+  when completed $ do
+    unless keepTemp $ cleanup transcriptDir
+    when keepTemp $ PT.putPrettyLn $
```

Comment:
If I understood correctly, if the transcript fails, then the work directory is automatically preserved, right?  In that case, we still want it to explain what happened + how to `ucm -codebase <transcriptDir>` for the user to poke around in it to debug.

Reply:
In various places in the run method 
https://github.com/unisonweb/unison/blob/e42267fc671004970683aac3488b511be3aa8cdd/parser-typechecker/src/Unison/Codebase/TranscriptParser.hs#L99

Errors exit prematurely with `die`

https://github.com/unisonweb/unison/blob/e42267fc671004970683aac3488b511be3aa8cdd/parser-typechecker/src/Unison/Codebase/TranscriptParser.hs#L235-L241

I think this can be accomplished by 
1. updating the `dir` passed to `run` to be the `transcriptDir`, so the error points to the correct temporary path
2. adding the output message for how to debug.

Reply:
Ok, great.  That `dir` on line 241 in your excerpt above, is that not already the `transcriptDir`?

Reply:
No, that is not the transcript dir, have taken care of it in the latest commit.

Reply:
Out of curiosity; which dir was it?  Sorry I’m just on github mobile site, not a full IDE in front of me

Reply:
It was the current directory . 

Originally a parse failure would fail with (when running from inside "test"):
```
🛑

Transcript failed due to the message above.

Run `ucm -codebase /.../test` to do more work with it
```

For `transcript` and `transcript.fork`, the new codebase has been initialised in the temporary directory, and copied in the temporary directory respectively, so the message was not correct.

Currently:
```
🛑

Transcript failed due to the message above.

Run `ucm -codebase /.../test/transcript-72911f80dd3db7c8` to do more work with it.
```

I am still missing a space in the string concatenation, just noticed.

Reply:
Whoops, this is much better, thanks!